parcels/field.py

```diff
@@ -536,6 +516,7 @@ def from_netcdf(
                 interp_method = interp_method[variable[0]]
             else:
                 raise RuntimeError(f"interp_method is a dictionary but {variable[0]} is not in it")
+        interp_method = cast(InterpMethodOption, interp_method)
```

Comment:
What does this do? is the idea that `InterpmethodOption` also has a `dtype` that it casts to?

Reply:
This is to tell mypy that `interp_method` is no longer `InterpMethod` but is now `InterpMethodOption`